scipy/optimize/_constraints.py

```diff
@@ -0,0 +1,429 @@
+                raise ValueError("`lb` has the wrong dimension.")
+            if keyword == "less":
+                raise ValueError("`ub` has the wrong dimension.")
+            if keyword == "equals":
+                raise ValueError("`c` has the wrong dimension.")
+        c = np.resize(c, m)
+        return (keyword, c)
+    elif keyword == "interval":
+        lb = np.asarray(kind[1], dtype=float)
+        if np.size(lb) not in (1, m):
+            raise ValueError("`lb` has the wrong dimension.")
+        lb = np.resize(lb, m)
+        ub = np.asarray(kind[2], dtype=float)
+        if np.size(ub) not in (1, m):
+            raise ValueError("`ub` has the wrong dimension.")
+        ub = np.resize(ub, m)
+        if (lb > ub).any():
+            raise ValueError("lb[i] > ub[i].")
+        return (keyword, lb, ub)
+
+
+def _check_enforce_feasibility(enforce_feasibility, m):
+    if isinstance(enforce_feasibility, bool):
+        enforce_feasibility = np.full(m,
+                                      enforce_feasibility,
+                                      dtype=bool)
+    else:
+        enforce_feasibility = np.array(enforce_feasibility,
+                                       dtype=bool)
+
+        if enforce_feasibility.size != m:
+            raise ValueError("The parameter 'enforce_feasibility' "
+                             "has the wrong number of elements.")
+    return enforce_feasibility
+
+
+def _is_feasible(kind, enforce_feasibility, f0):
+    keyword = kind[0]
+    if keyword == "equals":
+        lb = np.asarray(kind[1], dtype=float)
+        ub = np.asarray(kind[1], dtype=float)
+    elif keyword == "greater":
+        lb = np.asarray(kind[1], dtype=float)
+        ub = np.full_like(lb, np.inf, dtype=float)
+    elif keyword == "less":
+        ub = np.asarray(kind[1], dtype=float)
+        lb = np.full_like(ub, -np.inf, dtype=float)
+    elif keyword == "interval":
+        lb = np.asarray(kind[1], dtype=float)
+        ub = np.asarray(kind[2], dtype=float)
+    else:
+        raise RuntimeError("Never be here.")
+
+    return ((lb[enforce_feasibility] <= f0[enforce_feasibility]).all()
+            and (f0[enforce_feasibility] <= ub[enforce_feasibility]).all())
+
+
+def _reinforce_box_constraint(kind, enforce_feasibility, x0,
+                              relative_tolerance=0.01,
+                              absolute_tolerance=0.01):
+        """Reinforce box constraint"""
+        x0 = np.copy(np.asarray(x0, dtype=float))
+        keyword = kind[0]
+        if keyword == "greater":
+            lb = np.asarray(kind[1], dtype=float)
+            ub = np.full_like(lb, np.inf, dtype=float)
+        elif keyword == "less":
+            ub = np.asarray(kind[1], dtype=float)
+            lb = np.full_like(ub, -np.inf, dtype=float)
+        elif keyword == "interval":
+            lb = np.asarray(kind[1], dtype=float)
+            ub = np.asarray(kind[2], dtype=float)
+
+        x0_new = np.copy(x0)
```

Comment:
While not terribly important, but perhaps it's better to convert this loop into vectorized code. To start with you can simply select components with `enforce_feasibility`. But if it gets too complicated, then let's keep it as it is.

Reply:
I actually tried to do it in a vectorized way at first but it got too complicated. This is not a very critical part of the code (it runs only once) so I decided to do it using a loop.

I think it will be beter if we keep it as it is.